calico_rkt/calico_rkt.py

```diff
@@ -20,381 +20,474 @@
+        self._client.set_endpoint(endpoint)
+        _log.info("Provisioned veth for endpoint using netns path %s on "
+                  "interface %s", netns_path, self.interface)
+        return endpoint
+
+    def _set_profile_on_endpoint(self, endpoint, profile_name):
+        """Sets given profile_name on given endpoint.
+
+        If profile for given profile_name does not exist, create a profile.
+
+        :param endpoint
+        :param profile_name
+        :return: None.
+        """
+        if not self._client.profile_exists(profile_name):
+            _log.info("Creating new profile %s.", profile_name)
+            self._client.create_profile(profile_name)
+
+        _log.info("Setting profile %s on endpoint.", profile_name)
+        self._client.set_profiles_on_endpoint(profile_names=[profile_name],
+                                              endpoint_id=endpoint.endpoint_id)
 
-    # Remove the container from the datastore.
-    datastore_client.remove_workload(hostname=hostname,
-                                     orchestrator_id=orchestrator_id,
-                                     workload_id=container_id)
+    def _remove_profile(self, profile_name):
+        """Remove the profile from datastore if it has no endpoints attached.
+
+        :param profile_name: Name of profile to remove.
+        :return: None.
+        """
+        profile_members = self._client.get_profile_members(profile_name)
+        if self._client.profile_exists(profile_name) and len(profile_members) < 1:
+            try:
+                _log.info("Profile %s has no members, removing from datastore.",
+                          profile_name)
+                self._client.remove_profile(profile_name)
+            except KeyError:
+                _log.error("Cannot remove profile %s: Profile not found.",
+                           profile_name)
+                sys.exit(1)
+
+    def _get_endpoint(self):
+        """Gets endpoint matching the container_id.
+
+        Exits gracefully if no endpoint is found.
+        Exits with an error if multiple endpoints are found.
+
+        :param container_id:
+        :return: Calico endpoint object
+        """
+        try:
+            _log.info("Retrieving endpoint that matches container ID %s",
+                      self.container_id)
+            ep = self._client.get_endpoint(hostname=HOSTNAME,
+                                           orchestrator_id=ORCHESTRATOR_ID,
+                                           workload_id=self.container_id)
+        except KeyError:
+            _log.warning("No endpoint found matching ID %s", self.container_id)
+            sys.exit(0)
+        except MultipleEndpointsMatch:
+            _log.error("Multiple endpoints found matching ID %s", self.container_id)
+            sys.exit(1)
 
-    _log.info("Removed Calico interface from %s" % container_id)
+        return ep
 
+    def _find_ipam_plugin(self):
```

Comment:
This could be more generically useful if it was changed to `_find_plugin(self, plugin_name)`.

 It could then search the CNI_PATH for a plugin with the given name, and could be used for more than just IPAM plugins sometime in the future.  For now, the only type of plugin that exists is IPAM, so this is fine, but thought I'd mention it anyway.